src/data_utils/dataset.py

```diff
@@ -0,0 +1,102 @@
+from pathlib import Path
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+import torch
+from albumentations import (
+    Affine,
+    Blur,
+    CoarseDropout,
+    Compose,
+    Downscale,
+    GridDistortion,
+    Perspective,
+    ShiftScaleRotate,
+    Resize,
+)
+from albumentations.pytorch.transforms import ToTensorV2
+from torch.utils.data import Dataset
+
+
+class AlgalDataset(Dataset):
+    """Reads in an image, transforms pixel values, and serves
+    a dictionary containing the image tensors, and labels.
+    """
+
+    def __init__(
+        self,
+        data_dir: Path,
+        csv_path: Path,
+        phase: str,
+        augmentations_intensity: float = 0.0,
+        test_size: int = 0,
+    ):
+        self.data_dir = data_dir
+        self.images = data_dir.rglob("*.npy")
+        self.images_dict = defaultdict()
+        for image in self.images:
+            self.images_dict[image.stem] = image
+        df = pd.read_csv(csv_path)
+        df = df[df["split"] == phase]
+        self.data = df if test_size <= 0 else df.iloc[:test_size]
+        self.data["filepath"] = self.data.loc[:, "uid"].apply(lambda x: self.images_dict[x])
+        self.labels = self.data.loc[:, "severity"]
+
+        self.transform = Compose(
+            [
+                Resize(224, 224),
+                ToTensorV2(),
+            ]
+        )
+        self.augmentation = None
```

Comment:
[nit] move to separate function
you can also add parametrization for different augmentation types, e.g. "weak": <few augs>, "medium": \<more augs>, "hard": \<even more augs>

Reply:
I still haven't saw the data but pretty sure `fliplr`, `flipud` and `colorjitter` will be good additional choices for augmentations, consider adding them